src/integrator.jl

```diff
@@ -69,6 +75,36 @@ function step(
     end
     return z
 end
+# TODO: merge step and steps using Task
+function steps(
```

Comment:
@yebai Another main change is here.

Reply:
This implementation is pretty naive now. I guess we can unify `step` and `steps` by implementing the leapfrog using `Task`. But I'd leave it for another PR.

Reply:
I'm slightly confused here. It does not seem to add any new functionality. The old `step` function can handle both 1-step and multiple-step leapfrog integration.

Reply:
This function is to **return** all points during the integration.

Reply:
I see. Maybe we can support this via adding a keyword argument to the original function, e.g. 

```julia
function step(
    lf::AbstractLeapfrog{T},
    h::Hamiltonian,
    z::PhasePoint,
    n_steps::Int=1;
    fwd::Bool=n_steps > 0,  # simulate hamiltonian backward when n_steps < 0,
    zs::Union{Vector{PhasePoint}, PhasePoint} = z # change to Vector{PhasePoint}(undef, n_steps) otherwise.
) where {T<:AbstractScalarOrVec{<:AbstractFloat}}
    n_steps = abs(n_steps)  # to support `n_steps < 0` cases
    ϵ = fwd ? step_size(lf) : -step_size(lf)
    ϵ = ϵ'

    @unpack θ, r = z
    @unpack value, gradient = ∂H∂θ(h, θ)

    for i = 1:n_steps
        # Tempering
        r = temper(lf, r, (i=i, is_half=true), n_steps)
        # Single leapfrog step
        θ, r, value, gradient = step(lf, ϵ, h, θ, r, value, gradient)
        # Tempering
        r = temper(lf, r, (i=i, is_half=false), n_steps)
        # Create a new phase point by caching the logdensity and gradient
        z = phasepoint(h, θ, r; ℓπ=DualValue(value, gradient))
        !isfinite(z) && break

      if zs  isa Vector 
        zs[i] = z
      else 
        zs = z
      end

    end
    return zs
end
```

This would avoid the clutter of introducing two additional `step` functions.

Reply:
Using `zs` to name a single point is confusing. I have an implementation which is nice. I will push it later.

Reply:
Apparently the channel approach causes some unexpected errors which I cannot find the root. I will implement your version first for now.